inc/class-component.php

```diff
@@ -170,6 +178,37 @@ public function load_example_data( int $index ) {
 		}
 	}
 
+	/**
+	 * Show errors on WPCL admin screens only.
+	 *
+	 * @return void
+	 */
+	private function maybe_show_errors(): void {
+		if ( empty( $this->error_reasons ) || ! is_admin() ) {
+			return;
+		}
+
+		if ( apply_filters( 'wpcl_hide_component_errors_as_notices', false ) ) {
+			return;
+		}
+
+		foreach ( $this->error_reasons as $error ) {
+			add_action(
+				'wpcl_admin_notices',
+				function() use ( $error ) {
+					$error = "$this->name - $error";
+					wpcl_component(
+						'wpcl-admin-notice',
+						[
+							'type' => 'error',
+							'text' => $error,
+						]
+					);
+				}
+			);
```

Comment:
Although this works, what I would prefer is for there to be, say, a static class variable that can collect errors, and then you would have a generic action (tied to a named function) that would loop through those and output each. Basically, having anonymous callback functions for actions _works_ but is a huge pain to manage because you can't unhook them, so we prefer to use named callback functions for this.

Reply:
I don't disagree, but I also don't see the utility in this case since all errors can be turned off anyway. I'll make a todo for it and open an issue with a link back to this comment.